core/src/avm2/traits.rs

```diff
@@ -29,6 +31,22 @@ bitflags! {
     }
 }
 
+// Represents a single key-value pair for a trait metadata.
+#[derive(Clone, Collect, Debug, Eq, PartialEq)]
+#[collect(no_drop)]
+pub struct TraitMetadataItem<'gc> {
+    pub key: AvmString<'gc>,
+    pub value: AvmString<'gc>,
+}
+
+// Represents a single metadata item for a trait.
+#[derive(Clone, Collect, Debug, Eq, PartialEq)]
+#[collect(no_drop)]
+pub struct TraitMetadata<'gc> {
```

Comment:
Actually, what do you think about moving this code to `metadata.rs` and dropping the `Trait` from the struct name?
The structs themselves and `fn metadata_from_abc_trait` don't really have anything directly related to traits; and traits.rs in the past avoided directly interacting with TU.